src/test/java/org/sasanlabs/service/vulnerability/commandInjection/CommandInjectionTest.java

```diff
@@ -0,0 +1,119 @@
+
+    private static CommandInjection commandInjection;
+    private static final String successReturnStr = "PING SUCCESS";
+
+    @BeforeEach
+    void setUp() throws IOException {
+        commandInjection = Mockito.spy(new CommandInjection());
+
+        StringBuilder stringBuilder = new StringBuilder().append(successReturnStr);
+
+        // Real localhost ping is not able to operate in CI, Mock instead
+        doReturn(stringBuilder)
+                .when(commandInjection)
+                .getResponseFromPingCommand(eq(LOCALHOST), eq(true));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel1_Localhost_ExpectPingSuccess() throws IOException {
+        // Act
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel1(LOCALHOST);
+
+        // Assert
+        assertStatusAndValid(responseEntity);
+        assertThat(responseEntity.getBody().getContent()).isEqualTo(successReturnStr);
+
+        verify(commandInjection).getVulnerablePayloadLevel1(LOCALHOST);
+        verify(commandInjection).getResponseFromPingCommand(eq(LOCALHOST), eq(true));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel1_NoIp_ExpectNoResponseContent() throws IOException {
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel1(Strings.EMPTY);
+
+        // Assert
+        assertStatusAndValid(responseEntity);
+        assertThat(responseEntity.getBody().getContent()).isEqualTo(Strings.EMPTY);
+
+        verify(commandInjection).getVulnerablePayloadLevel1(eq(Strings.EMPTY));
+        verify(commandInjection).getResponseFromPingCommand(eq(Strings.EMPTY), eq(false));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel2_Localhost_ExpectPingSuccess()
+            throws ServiceApplicationException, IOException {
+        RequestEntity entity = new RequestEntity(HttpMethod.GET, URI.create(LOCALHOST));
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel2(LOCALHOST, entity);
+
+        assertStatusAndValid(responseEntity);
+        assertThat(responseEntity.getBody().getContent()).isEqualTo(successReturnStr);
+
+        verify(commandInjection).getVulnerablePayloadLevel2(eq(LOCALHOST), eq(entity));
+        verify(commandInjection).getResponseFromPingCommand(eq(LOCALHOST), eq(true));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel2_NoIp_ExpectNoResponseContent()
+            throws ServiceApplicationException, IOException {
+        RequestEntity entity = new RequestEntity(HttpMethod.GET, URI.create(Strings.EMPTY));
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel2(Strings.EMPTY, entity);
+
+        // Assert
+        assertStatusAndValid(responseEntity);
```

Comment:
instead of this method i think what you can do is, pass the boolean flag to a private method and when boolean is false response is same across all false conditions and when it is true response is same across all true conditions.